`igamma` and `igammac`: port to structured

Freey0 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #57630 Port maximum and minimum to structured
* #57629 Port logaddexp and logaddexp2 to structured
* #57628 Port lcm to structured
* #57627 Port hypot to structured
* **#57626 Port igamma and igammac to structured**
* #57625 Port nextafter to structured
* #57624 Port gcd to structured

Differential Revision: [D28224834](https://our.internmc.facebook.com/intern/diff/D28224834)

Freey0 committed:
Port igamma and igammac to structured

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28224834).

Freey0 committed:
Update on "Port igamma and igammac to structured"


Differential Revision: [D28224834](https://our.internmc.facebook.com/intern/diff/D28224834)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28224834).
